internal/service/directconnect/macsec_key_test.go

```diff
@@ -0,0 +1,125 @@
+package directconnect_test
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/directconnect"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-aws/internal/acctest"
+)
+
+func TestAccDirectConnectMacSecKey_withCkn(t *testing.T) {
+	// Requires an existing MACsec-capable DX connection set as environmental variable
+	key := "DX_CONNECTION_ID"
+	connectionId := os.Getenv(key)
+	if connectionId == "" {
+		t.Skipf("Environment variable %s is not set", key)
+	}
+	resourceName := "aws_dx_macsec_key.test"
+	ckn := testAccDirecConnectMacSecGenerateHex()
+	cak := testAccDirecConnectMacSecGenerateHex()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ErrorCheck:               acctest.ErrorCheck(t, directconnect.EndpointsID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             nil,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccMacSecConfig_withCkn(ckn, cak, connectionId),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "connection_id", connectionId),
+					resource.TestMatchResourceAttr(resourceName, "ckn", regexp.MustCompile(ckn)),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				// Ignore the "cak" attribute as isn't returned by the API during read/refresh
+				ImportStateVerifyIgnore: []string{"cak"},
+			},
+		},
+	})
+}
+
+func TestAccDirectConnectMacSecKey_withSecret(t *testing.T) {
+	// Requires an existing MACsec-capable DX connection set as environmental variable
+	dxKey := "DX_CONNECTION_ID"
+	connectionId := os.Getenv(dxKey)
+	if connectionId == "" {
+		t.Skipf("Environment variable %s is not set", dxKey)
+	}
+
+	secretKey := "SECRET_ARN"
+	secretArn := os.Getenv(secretKey)
+	if secretArn == "" {
+		t.Skipf("Environment variable %s is not set", secretKey)
+	}
+
+	resourceName := "aws_dx_macsec_key.test"
```

Comment:
Have you considered naming this resource `aws_dx_macsec_key_association`? Since MACsec keys can only be associated/described/dissociated, renaming the resource may help preventing any confusion. 
```suggestion
	resourceName := "aws_dx_macsec_key_association.test"
```